### PR TITLE
Capistrano recipe

### PR DESCRIPTION
**Pull request by javierg**
Probably we want to move config/application.yml to a different folder in the server and symlink to shared, we don't want this data to be on the repo.

taking care of #32

### CODE REVIEW

**Comment by javierg** [comment]
Need to fix some issue with symlinking shared/config/database.yml -> config/database.yml

**Comment by javierg** [comment]
Symlinks added to config files, also, for CI deployment, need to create a host called live.rt that points to the right IP.

I am seeing the Invalid credentials error on this server also as in local, will like if some one can test it.

File: lib/tasks/config.rake [hunk 1]
```diff
@@ -7,8 +7,8 @@ namespace :config do
     require 'erb'
 
     erb_env = Proc.new do
-      @host = Rails.env.development? ? 'www.rt.dev' : 'www.rubitij.org'
-      @root = Rails.env.development? ? Rails.root : '/home/deployer/app'
+      @host = Rails.env.development? ? 'www.rt.dev' : 'www.rubytij.org'
+      @root = Rails.root
```
Comment (javierg) [inline lib/tasks/config.rake]: probably will just add the static path for production...?

File: Capfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+load 'deploy' if respond_to?(:namespace)
+Dir['vendor/gems/*/recipes/*.rb','vendor/plugins/*/recipes/*.rb'].each { |plugin| load(plugin) }
```
Comment (EnriqueVidal) [inline Capfile:4]: Why are we doing this? we have no plugins nor gems in vendor, actually `vendor/bundle` should be shared too. possibly autogen-code?

File: config/deploy.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+role :web, "live.rt"
+role :db, "live.rt", :primary => true
+
+namespace :deploy do
+  task :config, :roles => [:app, :web] do
+    run "ln -nfs #{shared_path}/config/database.yml #{release_path}/config/database.yml"
+    run "ln -nfs #{shared_path}/config/application.yml #{release_path}/config/application.yml"
+  end
+  task :start do ; end
+  task :stop do ; end
```
Comment (EnriqueVidal) [inline config/deploy.rb]: Won't this override the exiting start and stop tasks?

File: config/deploy.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  task :restart, :roles => :app do
+    run "#{try_sudo} touch #{File.join(current_path,'tmp','restart.txt')}"
```
Comment (EnriqueVidal) [inline config/deploy.rb:23]: I don't think we need sudo, the app runs with the same user it's fetched as.

**Comment by EnriqueVidal** [comment]
I couldn't find this on the recipe, is the recipe running `bundle install --deployment --without development test`?

File: Capfile (hunk 2 above)
Comment (javierg) [inline Capfile:4]: just the default Capfile :)

File: config/deploy.rb (hunk 3 above)
Comment (javierg) [inline config/deploy.rb]: again, default cap deploy generated file... I can remove those...

File: config/deploy.rb (hunk 4 above)
Comment (javierg) [inline config/deploy.rb:23]: ok, just remove it

**Comment by javierg** [comment]
yes it is, is the capistrano/bundler included

> I couldn't find this on the recipe, is the recipe running `bundle install --deployment --without development test`?

**Comment by javierg** [comment]
Y U NO merge me!

**Comment by javierg** [comment]
ok, keep failing on travis... :checking:

File: config/deploy.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+set :user, "deployer"
+set :ssh_name, "live.rt"
+ssh_options[:forward_agent] = true
+set :deploy_to, "/home/#{ user }/srv/app/#{application}"
+set :deploy_via, :remote_cache
+set :rails_env, ( ENV['RAILS_ENV'] = 'production' )
+
+role :app, "live.rt"
+role :web, "live.rt"
+role :db, "live.rt", :primary => true
```
Comment (EnriqueVidal) [inline config/deploy.rb]: I feel this is flat-out unnecessary we could just use our public domain name `rubytij.org` no need for the extra config.

**Comment by EnriqueVidal** [comment]
Working on this, I noticed it does not create the `releases` folder on cold deploy.

File: config/deploy.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  task :cold do
+    run "mkdir -p #{ deploy_to }/releases"
+    run "mkdir -p #{ shared_path }/log"
+
+    update
+  end
+
+  task :gems do
```
Comment (EnriqueVidal) [inline config/deploy.rb]: Hmmm... maybe we should rename this to bundler, any ideas @javierg ?

**Comment by javierg** [comment]
we don;t need the cold deploy, it is done by `cap deploy:setup`

**Comment by EnriqueVidal** [comment]
`config:generate` needs updating, I'll do this this weekend for now, the multi-staging deploy is ready, as soon as we have a staging server again I'll make our first deploy.

**Pull request merged**